### PR TITLE
1.6 CI fixes

### LINKED ISSUE #8775: [v1.7] CI: K8sIstioTest Istio Bookinfo Demo Tests bookinfo inter-service connectivity
## CI failure

```
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/k8s-1.10-gopath/src/github.com/cilium/cilium/test/ginkgo-ext/scopes.go:420
Pods are not ready after timeout
Expected
    <*errors.errorString | 0xc0004eeb60>: {
        s: "Timeout reached: timed out waiting for pods with filter -l zgroup=bookinfo to be ready",
    }
to be nil
/home/jenkins/workspace/Ginkgo-CI-Tests-Pipeline/k8s-1.10-gopath/src/github.com/cilium/cilium/test/k8sT/istio.go:290
```

https://jenkins.cilium.io/job/Ginkgo-CI-Tests-Pipeline/7972/testReport/Suite-k8s-1/10/K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter_service_connectivity/

[88d41cea_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip](https://github.com/cilium/cilium/files/3467594/88d41cea_K8sIstioTest_Istio_Bookinfo_Demo_Tests_bookinfo_inter-service_connectivity.zip)

### PR DESCRIPTION
**Pull request by tgraf**
Various fixes to address 1.6 CI stability

Fixes: #8775 

---
This change is [Reviewable](https://reviewable.io/reviews/cilium/cilium/8779)

### CODE REVIEW

**Comment by tgraf** [comment]
test-me-please

**Review by nebril — APPROVED** [review]

**Comment by tgraf** [comment]
Running a 2nd CI run as this can potentially resolves #8780, #8778, #8777, #8776

**Comment by tgraf** [comment]
test-me-please

File: test/helpers/kubectl.go [hunk 1]
```diff
@@ -1013,52 +1032,63 @@ func addIfNotOverwritten(options []string, field, value string) []string {
...
+
+	res := kub.Apply("cilium.yaml")
 	if !res.WasSuccessful() {
 		return res.GetErr("Unable to apply YAML")
 	}
 
 	return nil
 }
 
+// ciliumUninstallHelm installs Cilium with the Helm options provided.  Returns an
```
Comment (ianvernon) [inline test/helpers/kubectl.go]: installs --> uninstalls

File: test/helpers/kubectl.go [hunk 2]
```diff
@@ -1013,52 +1032,63 @@ func addIfNotOverwritten(options []string, field, value string) []string {
...
 	if !res.WasSuccessful() {
 		return res.GetErr("Unable to generate YAML")
 	}
 
-	res = kub.Apply("cilium.yaml")
+	return nil
+}
+
+// ciliumInstallHelm installs Cilium with the Helm options provided.  Returns an
+// error if any patch or if any original descriptors files were not found.
```
Comment (ianvernon) [inline test/helpers/kubectl.go]: Documentation seems to be leftover from the prior Cilium installation methods?

File: test/helpers/kubectl.go [hunk 3]
```diff
@@ -1013,52 +1032,63 @@ func addIfNotOverwritten(options []string, field, value string) []string {
...
 }
 
+// ciliumUninstallHelm installs Cilium with the Helm options provided.  Returns an
+// error if any patch or if any original descriptors files were not found.
+func (kub *Kubectl) ciliumUninstallHelm(options []string) error {
+	if err := kub.generateCiliumYaml(options, "cilium.yaml"); err != nil {
+		return err
+	}
+
+	res := kub.Delete("cilium.yaml")
```
Comment (ianvernon) [inline test/helpers/kubectl.go:1071]: Do we generate different resources in K8s based off of the Helm options? Most of these appear to be options provided to the DaemonSet / ConfigMap - can you elaborate on what K8s resources are different depending upon the Helm options?

Comment (tgraf) [reply test/helpers/kubectl.go:1071]: * Inclusion of etcd-operator and related ClusterRole, ClusterRoleBinding and ServiceAccount.

**Comment by tgraf** [comment]
test-me-please

**Comment by tgraf** [comment]
@nebril This seems to fix #8775 and maybe also #8777

**Pull request merged**